aclogview/CM_Communication.cs

```diff
@@ -68,4 +176,123 @@ public class WeenieErrorWithString : Message {
             treeView.Nodes.Add(rootNode);
         }
     }
+
+    public class HearSpeech : Message
+    {
+        public PStringChar MessageText;
+        public PStringChar SenderName;
+        public uint SenderID;
+        public uint ChatMessageType;
+
+        public static HearSpeech read(BinaryReader binaryReader)
+        {
+            var newObj = new HearSpeech();
+            newObj.MessageText = PStringChar.read(binaryReader);
+            newObj.SenderName = PStringChar.read(binaryReader);
+            newObj.SenderID = binaryReader.ReadUInt32();
+            newObj.ChatMessageType = binaryReader.ReadUInt32();
+            return newObj;
+        }
+
+        public override void contributeToTreeView(TreeView treeView)
+        {
+            TreeNode rootNode = new TreeNode(this.GetType().Name);
+            rootNode.Expand();
+            rootNode.Nodes.Add("MessageText = " + MessageText.m_buffer);
+            rootNode.Nodes.Add("SenderName = " + SenderName.m_buffer);
+            rootNode.Nodes.Add("SenderID = " + SenderID);
+            rootNode.Nodes.Add("ChatMessageType = " + ChatMessageType);
+            treeView.Nodes.Add(rootNode);
+        }
+    }
+
+    public class HearRangedSpeech : Message
+    {
+        public PStringChar MessageText;
+        public PStringChar SenderName;
+        public uint SenderID;
+        public float Range;
+        public uint ChatMessageType;
+
+        public static HearRangedSpeech read(BinaryReader binaryReader)
+        {
+            var newObj = new HearRangedSpeech();
+            newObj.MessageText = PStringChar.read(binaryReader);
+            newObj.SenderName = PStringChar.read(binaryReader);
+            newObj.SenderID = binaryReader.ReadUInt32();
+            newObj.Range = binaryReader.ReadSingle();
+            newObj.ChatMessageType = binaryReader.ReadUInt32();
+            return newObj;
+        }
+
+        public override void contributeToTreeView(TreeView treeView)
+        {
+            TreeNode rootNode = new TreeNode(this.GetType().Name);
+            rootNode.Expand();
+            rootNode.Nodes.Add("MessageText = " + MessageText.m_buffer);
+            rootNode.Nodes.Add("SenderName = " + SenderName.m_buffer);
+            rootNode.Nodes.Add("SenderID = " + SenderID);
+            rootNode.Nodes.Add("Range = " + Range);
+            rootNode.Nodes.Add("ChatMessageType = " + ChatMessageType);
+            treeView.Nodes.Add(rootNode);
+        }
+    }
+
+    public class HearDirectSpeech : Message
+    {
+        public PStringChar MessageText;
+        public PStringChar SenderName;
+        public uint SenderID;
+        public uint TargetID;
+        public uint ChatMessageType;
+        public uint Unknown;

```

Comment:
Note that this unknown is "secretFlags", which oddly seems to not be used at all in the client.